VisualizationBase/src/Scene.h

```diff
@@ -244,7 +246,8 @@ inline Cursor* Scene::mainCursor() { return mainCursor_; }
 inline const QList<Item*>& Scene::topLevelItems() const {return topLevelItems_; }
 inline void Scene::addRefreshActionFunction(RefreshActionFunction func) {refreshActionFunctions_.append(func); }
 
-inline void Scene::addOnZoomHandler(OnZoomHandler onZoomHandler) {onZoomHandlers_.append(onZoomHandler);}
+inline void Scene::addOnZoomHandler(OnZoomHandler onZoomHandler, OnZoomHandlerRemove onZoomHandlerRemove)
```

Comment:
So, instead of `void`, this function could return some sort of a handle, or an index, which you can then later pass to the `removeOnZoomHandler` function to remove a specific handler. Perhaps store the handlers in a `QMap` and give each newly regitsred handle an unique id (increment a static int).